ipa-core/src/protocol/prss/crypto.rs

```diff
@@ -24,16 +27,38 @@ pub trait FromRandomU128 {
     fn from_random_u128(src: u128) -> Self;
 }
 
-/// Trait for things that can be generated by PRSS.
+impl FromRandomU128 for u128 {
+    fn from_random_u128(src: u128) -> Self {
+        src
+    }
+}
+
+/// Trait for random generation.
 ///
 /// The exact semantics of the generation depend on the value being generated, but like
 /// `rand::distributions::Standard`, a uniform distribution is typical. When implementing
 /// this trait, consider the consequences if the implementation were to be used in
 /// an unexpected way. For example, an implementation that draws from a subset of the
 /// possible values could be dangerous, if used in an unexpected context where
 /// security relies on sampling from the full space.
+pub trait FromRandom: Sized {
+    type SourceLength: ArrayLength;
+
+    /// Generate a random value of `Self` from `SourceLength` uniformly-distributed u128s.
+    fn from_random(src: GenericArray<u128, Self::SourceLength>) -> Self;
```

Comment:
curious about the choice of `u128` for container? why not `u8`?

Reply:
I used `u128` because it matches what we get from PRSS. I'd be fine with using `u8`. The only hesitation I have about `u8` is that it hides the fact that the randomness is generated in multiples of 128 bits, i.e., an implementation that takes a single `u8` is much less efficient than it looks.

Reply:
agree, it is better to indicate that it is a multiple of 16 bytes